molecule/shared/cleanup.yml

```diff
@@ -0,0 +1,19 @@
+- name: cleanup
+  gather_facts: false
+  hosts: rhsm
+  tasks:
+    - name: Check if host to clean up is available
+      wait_for_connection:
+        # connect_timeout defaults to 5 seconds, so setting
+        # timeout to less than that should give one connection attempt
+        timeout: 1
+      register: wait_result
+      ignore_errors: true
+
+    - name: Unregister system if available
+      include_role:
+        name: rhsm
+      vars:
+        rhsm_username: "{{ omit }}"
+        rhsm_unregister: true
+      when: wait_result is success
```

Comment:
I just complicated this one a bit more in the meta_skeleton role to account for times when we're on a non-RH system. I guess that's not as relevant for this role, is it?